### DISCUSSION TITLE
Why non-visible item in list store, clears its state?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#3707] [Q&A] [answered]
Looks like bug, but maybe is there switch to  enable "proper behavior"

https://github.com/slint-ui/slint/assets/41945903/02e975d4-1899-42f5-b1a8-3919111aaa57

**Comment by ogoffart** [comment, marked as answer]
Yes, the state is cleared as items gets recreated.

The fix is to keep the checked and selection state as part of the model.
